tools/check_meson.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+"""
+Script to check if all python sources files are actually registered in meson
+"""
+import os
+import subprocess
+import sys
+
+
+
+
+def get_all_files(root):
+    """list of all files in the git repo:
+
+    :param root: root of the project
+    :return: list of files in the repo
+    """
+    process = subprocess.run(["git","ls-files"], cwd=root, stdout=subprocess.PIPE)
+    if process.returncode !=0:
+        print("`git` command failed")
+    res = process.stdout.decode()
+    return  [i.strip() for i in res.split(os.linesep)]
+
+def main(root):
+    """print out all python filenames not properly registered in meson.build
+
+    :param root: path of the root of the project
+    :return: number of missing files/directories.
+    """
+    cnt = 0
+    src = []
+    meson = []
+
+    for line in get_all_files(root):
+        if line.endswith(".py") and line.startswith("src"):
+            src.append(line)
+        elif line.endswith("meson.build"):
+            meson.append(line)
```

Comment:
This check does not only apply to Python files but to all files (.png, .svg, .cl, .c, .pyx, ....) except C/C++/Cython headers.
To check everything, all files should be included but ".h", ".hpp" and ".pxd" extensions:

```suggestion
        if line.endswith("meson.build"):
            meson.append(line)
        else:
            extension = os.path.splitext(line)[1]
            if line.startswith("src") and extension not in (".h", ".hpp", ".pxd"):
                src.append(line)
```

Actually, trying this, I found a missing file: `src/silx/math/combo.pyi` and a `README` in histogramnd that I'm not sure what to do with.

It also needs to handle C files declared in a `meson.build` in a parent folder.. I'll continue this tomorrow

Reply:
Those files are never used from any other module, so they don't deserve installation, nor the c, h, cpp, hpp,  pyx, pxd, pxi, ... files. But this is my interpretation.

Reply:
> Those files are never used from any other module, so they don't deserve installation

I agree for the `README` (BTW it probably shouldn't be there) but not for the rest: Resource files (icons, OpenCL kernels,...) are required to be installed and so should be checked.
The `combo.pyi` type stub as well according to [PEP561](https://peps.python.org/pep-0561/#stub-only-packages):

> For package maintainers wishing to ship stub files containing all of their type information, it is preferred that the *.pyi stubs are alongside the corresponding *.py files.

Reply:
BTW, numpy has a similar check: https://github.com/numpy/numpy/blob/main/tools/check_installed_files.py